support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/TicketSerializer.java

```diff
@@ -0,0 +1,56 @@
+package org.apereo.cas;
+
+import org.apereo.cas.ticket.ServiceTicket;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketGrantingTicket;
+
+/**
+ * This is {@link TicketSerializer}.
+ *
+ * Abstraction layer for ticket serialization methods
+ *
+ * @author David Rodriguez
+ *
+ * @param <T> should be something that can be written
+ * to Cassandra like {@link String}
+ *
+ * @since 5.1.0
+ */
+public interface TicketSerializer<T> {
```

Comment:
This design here, while perfectly found, will not work because you're making certain upfront assumptions about the types of tickets that are found in the runtime. You need to be able to dynamically discover ticket types that exist and proceed to serialize those.